### PR TITLE
prepare capi support

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR sets some things so the CAPI support PR is easier to digest

### CODE REVIEW

File: helm/logging-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -37,7 +37,10 @@ spec:
         {{- else -}}
           - -enable-logging=false
         {{- end }}
-        - -installation-name={{ .Values.loggingOperator.installationName }}
+        - -installation-name={{ .Values.managementCluster.name }}
+        - -installation-provider={{ .Values.managementCluster.provider.kind }}
+        - -installation-region={{ .Values.aws.region }}
+        - -installation-base-domain={{ .Values.managementCluster.baseDomain }}
```
Comment (QuentinBisson) [inline helm/logging-operator/templates/deployment.yaml:43]: This is needed for IRSA

File: helm/logging-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -37,7 +37,10 @@ spec:
         {{- else -}}
           - -enable-logging=false
         {{- end }}
-        - -installation-name={{ .Values.loggingOperator.installationName }}
+        - -installation-name={{ .Values.managementCluster.name }}
+        - -installation-provider={{ .Values.managementCluster.provider.kind }}
+        - -installation-region={{ .Values.aws.region }}
```
Comment (QuentinBisson) [inline helm/logging-operator/templates/deployment.yaml:42]: THis is needed to create an IAM client

File: helm/logging-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -37,7 +37,10 @@ spec:
         {{- else -}}
           - -enable-logging=false
         {{- end }}
-        - -installation-name={{ .Values.loggingOperator.installationName }}
+        - -installation-name={{ .Values.managementCluster.name }}
+        - -installation-provider={{ .Values.managementCluster.provider.kind }}
```
Comment (QuentinBisson) [inline helm/logging-operator/templates/deployment.yaml:41]: Needed to be able to create the loki role on capa MCs only

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,6 @@
 module github.com/giantswarm/logging-operator
 
-go 1.18
+go 1.20
```
Comment (QuentinBisson) [inline go.mod:3]: Bumps dependencies

File: helm/logging-operator/values.yaml [hunk 5]
```diff
@@ -2,6 +2,11 @@ global:
   podSecurityStandards:
     enforced: false
 
+aws:
+  accessKeyID: accesskey
```
Comment (QuentinBisson) [inline helm/logging-operator/values.yaml:6]: Needed to be able to connect to the aws api

File: helm/logging-operator/templates/deployment.yaml [hunk 6]
```diff
@@ -37,7 +37,10 @@ spec:
         {{- else -}}
           - -enable-logging=false
         {{- end }}
-        - -installation-name={{ .Values.loggingOperator.installationName }}
+        - -installation-name={{ .Values.managementCluster.name }}
```
Comment (hervenicol) [inline helm/logging-operator/templates/deployment.yaml:40]: This requires a change in `config` repo.

Comment (QuentinBisson) [reply helm/logging-operator/templates/deployment.yaml:40]: Yep it's done

Comment (QuentinBisson) [reply helm/logging-operator/templates/deployment.yaml:40]: Let me track it After lunch

Comment (QuentinBisson) [reply helm/logging-operator/templates/deployment.yaml:40]: Sée https://github.com/giantswarm/config/blob/b95dba75f82525f9ece4141b4eb312d4f3c4902e/default/apps/logging-operator/configmap-values.yaml.template#L2

File: helm/logging-operator/templates/deployment.yaml (hunk 2 above)
Comment (hervenicol) [reply helm/logging-operator/templates/deployment.yaml:42]: "region" is probably AWS-specific.
Maybe that should be made clear in its name?

Comment (QuentinBisson) [reply helm/logging-operator/templates/deployment.yaml:42]: Well it is not really as it's also set in azure but I need to move where it is defined in config

File: pkg/capicluster/cluster.go [hunk 7]
```diff
@@ -14,12 +18,24 @@ type Object struct {
...
-	return value
+	loggingLabelValue, ok := labels[key.LoggingLabel]
+	if !ok {
+		return true
+	}
+
+	loggingEnabled, err := strconv.ParseBool(loggingLabelValue)
+	if err != nil {
+		return false
+	}
```
Comment (hervenicol) [inline pkg/capicluster/cluster.go:37]: If there is no label, it returns `true` => this means logging becomes enabled by default, right?
But if there's an error in `ParseBool`, it returns `false` => why not sticking to the default behaviour (`true`) then?

Comment (QuentinBisson) [reply pkg/capicluster/cluster.go:37]: Makes sense

Comment (QuentinBisson) [reply pkg/capicluster/cluster.go]: Fixed

**Review by hervenicol — APPROVED** [review]
I'm not very happy that it has to access AWS api, but apart from that LGTM.

**Pull request merged**